### ISSUE TITLE
Week 2 tutorial section 4.4

### ISSUE THREAD

**Issue by PhaedraKS** [BlueBrain/MOOC-neurons-and-synapses-2017#98]
Hello,
The section 4.4 of tutorial week 2 says to: "Then create a Dashboard, using the SPARQL query above."
I managed to make a new dashboard area and it has that query space so I put this in the dashboard query:
________________________________________________________________
PREFIX nxv: <https://bluebrain.github.io/nexus/vocabulary/>
PREFIX nsg: <https://neuroshapes.org/>
PREFIX schema: <http://schema.org/>
PREFIX prov: <http://www.w3.org/ns/prov#>
PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX skos: <http://www.w3.org/2004/02/skos/core#>
SELECT DISTINCT ?self ?name  ?brainRegion ?brainRegionLayer ?subjectSpecies ?subjectStrain ?registered_by ?registeredAt
WHERE {
?entity nxv:self                                            ?self ;
        a                                                   nsg:NeuronMorphology ;
        schema:name                                         ?name ;
        nxv:createdBy                                       ?registered_by ;
        nxv:createdAt                                       ?registeredAt ;
        nxv:deprecated                                      false ;
        nsg:brainLocation / nsg:brainRegion / rdfs:label    ?brainRegion ;
        nsg:brainLocation / nsg:layer                       ?brainRegionLayer ;
        nsg:subject / nsg:species / rdfs:label              ?subjectSpecies ;
        nsg:subject / nsg:strain / rdfs:label               ?subjectStrain.
}
LIMIT 1000
________________________________________________________________

But there is not anything in the Dashboard still after that. This is how it is looking:
![image](https://user-images.githubusercontent.com/97569886/165866895-a04ad682-4f27-4e7b-9ab5-a8384f3305b6.png)

And here is the querying in the dashboard:
![image](https://user-images.githubusercontent.com/97569886/165867047-c5b50b52-5bbb-4960-bc2b-0189beac6c07.png)

If I am meant to be putting some other query in there or anything else tricky,
please let me know,
thank you
Phaedra

**Comment by jdcourcol** [comment]
@MFSY @samuel-kerrien , can one of you look at this issue ?

**Comment by MFSY** [comment]
Hi @jdcourcol ,

I (or DKE) did not write this part of the tutorial. This is on @samuel-kerrien's team

**Comment by samuel-kerrien** [comment]
Dear Phaedra, 

Thanks a lot for getting in touch, sorry your are encountering issues in the tutorial.
I had a quick look at your query and it appears the formatting is different from the one in the tutorial page, in particular, it lacks the <> around the URI of the prefixes. Small details but the database likes it with them ;)

Please try to replace your query in the Studio Dashboard with the one below, it should work.

```
PREFIX nxv: <https://bluebrain.github.io/nexus/vocabulary/>
PREFIX nsg: <https://neuroshapes.org/>
PREFIX schema: <http://schema.org/>
PREFIX prov: <http://www.w3.org/ns/prov#>
PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
PREFIX skos: <http://www.w3.org/2004/02/skos/core#>

SELECT DISTINCT ?self ?name  ?brainRegion ?brainRegionLayer ?subjectSpecies ?subjectStrain ?registered_by ?registeredAt
WHERE {
?entity nxv:self                                   ?self ;
        a                                                   nsg:NeuronMorphology ;
        schema:name                             ?name ;
        nxv:createdBy                            ?registered_by ;
        nxv:createdAt                             ?registeredAt ;
        nxv:deprecated                          false ;
        nsg:brainLocation / nsg:brainRegion / rdfs:label    ?brainRegion ;
        nsg:brainLocation / nsg:layer                  ?brainRegionLayer ;
        nsg:subject / nsg:species / rdfs:label     ?subjectSpecies ;
        nsg:subject / nsg:strain / rdfs:label         ?subjectStrain.
}
LIMIT 1000
```

Please confirm if this solved your problem.

Cheers
Samuel

**Comment by PhaedraKS** [comment]
Hello Samuel,
Thank you for your suggestion and your help, am I using the right tutorial page? https://bluebrainnexus.io/docs/getting-started/try-nexus.html#4-2-project-and-resource-views
I did copy the query from there and I tried copying it from your post as well.

I am not sure what these different options mean because the tutorial doesn't seem to say which one to use:
![image](https://user-images.githubusercontent.com/97569886/166437414-e683a8ff-38c5-4f66-b4b8-7aa56ae464f3.png)



This did come up with a list:
![image](https://user-images.githubusercontent.com/97569886/166437647-033716a5-597a-4b09-ac1a-ae5e98d85150.png)



And I tried this other option:
![image](https://user-images.githubusercontent.com/97569886/166437942-76427b1f-1dfc-46a6-bd0c-c54580ab85a6.png)



Resulted in this:
![image](https://user-images.githubusercontent.com/97569886/166437844-4f29a9b5-4b60-4681-86d7-9a0c9f3547f5.png)



And this one doesn't end up with anything as a result:
![image](https://user-images.githubusercontent.com/97569886/166438954-08f5f686-fd31-434b-8fa1-df5145f975df.png)


These results aren't looking as they should according to the tutorial.
I hope that this issue can be fixed some how.
Thank you,
Phaedra

**Comment by samuel-kerrien** [comment]
Dear Phaedra,

Thanks for your further explanation, I see now there is some critical information missing in the tutorial.
You are indeed looking at the correct tutorial:
- the query is taken from here: https://bluebrainnexus.io/docs/getting-started/try-nexus.html#4-3-query-neuroscience-data
- the explanation of how to create the Studio is here: https://bluebrainnexus.io/docs/getting-started/try-nexus.html#4-4-create-a-studio

Now, what I realise is that there is a screen capture missing that shows how to create the Dashboard, and as you point out, using which data source. I include it here below in the meantime:

<img width="1437" alt="Screenshot 2022-05-03 at 18 16 29" src="https://user-images.githubusercontent.com/3679766/166493392-b7eb5872-1977-4b53-81bd-8f972ce43b70.png">

So in short, you have to pick the View named `defaultSparqlIndex`, which is the graph database that runs as part of Nexus. And the query you type in the field below (which you copy from [here](https://bluebrainnexus.io/docs/getting-started/try-nexus.html#4-3-query-neuroscience-data)), is the language that allows us to extract information about your datasets and lay it out as a more user friendly table.

I will make sure that we update the tutorial as soon as possible to that other users do benefit from what you have discovered. Thanks for helping making it better !

Best regards
Samuel

**Comment by PhaedraKS** [comment]
Thank you very much Samuel,
that is great that the tutorial will be updated with this critical information for others to benefit.
So, with the defaultSparqlIndex I end up with this:

![image](https://user-images.githubusercontent.com/97569886/166613243-ee144dcb-e41b-4b4d-9bfb-e2cad38d75f9.png)

Is there one more step missing somewhere that I need to do to somehow get everything organised under the headings,
I am confused a bit still because it's not coming up with registered_by, brainRegion, subjectSpecies etc.
I can see that those columns are in the query though. It's just resulting in one big list for some reason.

Thanks for all the help so far,
Phaedra

**Comment by PhaedraKS** [comment]
Hurray! 
I actually figured out what was happening and I have it the way it should be!

![image](https://user-images.githubusercontent.com/97569886/166614565-6b3abb62-010a-43cf-8fbb-b66566fa9776.png)

There was also no instruction about this preview button but I was just trying anything and suddenly its working properly:

![image](https://user-images.githubusercontent.com/97569886/166614768-39241e33-707f-4390-9398-2a24fe03829a.png)

So, students also need to be instructed to use the preview button and what to select, maybe even relabel the button to make people think that it does something important for organizing the information.
This issue is now complete, thank you for support,
Best,
Phaedra

**Comment by samuel-kerrien** [comment]
Hi Phaedra,

Glad you figured it out, looks like you have the Studio created in the end.
A few comments along the way:

- In [this message](https://github.com/BlueBrain/MOOC-neurons-and-synapses-2017/issues/98#issuecomment-1116863057), when I look at the first screen capture, I am guessing you have created the dashboard with the default SPARQL query that only shows one column. 
- Now in your [last message](https://github.com/BlueBrain/MOOC-neurons-and-synapses-2017/issues/98#issuecomment-1116869424), you took the SPARQL query from section 4.3 and set it in the dashboard configuration panel. This is how you got the desired output. I believe the preview button can be used to tweak the rendering of the table (enabling searching, sorting...) but is not essential in getting the data presented.

I am very sorry if this wasn't as easy as it should be, we will also learn from this experience and make it better for future learners.
Thanks a lot for investing your time, we really appreciate it!!
Best regards
Samuel
